src/components/transactions/TxInfo/index.tsx

```diff
@@ -84,56 +87,67 @@ export const TransferTx = ({
   return <></>
 }
 
-const CustomTx = ({ info }: { info: Custom }): ReactElement => {
-  return <>{info.methodName}</>
+const CustomTx = ({ info, truncateText }: { info: Custom; truncateText?: boolean }): ReactElement => {
+  return <Box className={classNames({ [css.txInfo]: truncateText })}>{info.methodName}</Box>
 }
 
-const CreationTx = ({ info }: { info: Creation }): ReactElement => {
-  return <>Created by {shortenAddress(info.creator.value)}</>
+const CreationTx = ({ info, truncateText }: { info: Creation; truncateText?: boolean }): ReactElement => {
+  return (
+    <Box className={classNames({ [css.txInfo]: truncateText })}>Created by {shortenAddress(info.creator.value)}</Box>
+  )
 }
 
-const MultiSendTx = ({ info }: { info: MultiSend }): ReactElement => {
+const MultiSendTx = ({ info, truncateText }: { info: MultiSend; truncateText?: boolean }): ReactElement => {
   return (
-    <>
+    <Box className={classNames({ [css.txInfo]: truncateText })}>
       {info.actionCount} {`action${info.actionCount > 1 ? 's' : ''}`}
-    </>
+    </Box>
   )
 }
 
-const SettingsChangeTx = ({ info }: { info: SettingsChange }): ReactElement => {
+const SettingsChangeTx = ({ info, truncateText }: { info: SettingsChange; truncateText?: boolean }): ReactElement => {
   if (
     info.settingsInfo?.type === SettingsInfoType.ENABLE_MODULE ||
     info.settingsInfo?.type === SettingsInfoType.DISABLE_MODULE
   ) {
-    return <>{info.settingsInfo.module.name}</>
+    return <Box className={classNames({ [css.txInfo]: truncateText })}>{info.settingsInfo.module.name}</Box>
   }
 
   return <></>
 }
 
-const TxInfo = ({ info, ...rest }: { info: TransactionInfo; omitSign?: boolean; withLogo?: boolean }): ReactElement => {
+const TxInfo = ({
+  info,
+  truncateText,
+  ...rest
+}: {
+  info: TransactionInfo
+  omitSign?: boolean
+  withLogo?: boolean
+  truncateText?: boolean
+}): ReactElement => {
   if (isSettingsChangeTxInfo(info)) {
-    return <SettingsChangeTx info={info} />
+    return <SettingsChangeTx info={info} truncateText={truncateText} />
   }
 
   if (isMultiSendTxInfo(info)) {
-    return <MultiSendTx info={info} />
+    return <MultiSendTx info={info} truncateText={truncateText} />
   }
 
   if (isTransferTxInfo(info)) {
     return <TransferTx info={info} {...rest} />
   }
 
   if (isCustomTxInfo(info)) {
-    return <CustomTx info={info} />
+    return <CustomTx info={info} truncateText={truncateText} />
   }
 
   if (isCreationTxInfo(info)) {
-    return <CreationTx info={info} />
+    return <CreationTx info={info} truncateText={truncateText} />
   }
 
   if (isOrderTxInfo(info)) {
-    return <SwapTx info={info} />
+    return <SwapTx info={info} truncateText={truncateText} />
```

Comment:
Can we avoid parametrizing all these components? Are there cases where we _don't_ want to truncate the text?

Reply:
Found that Apart from Pending Transaction section in home page <TxInfo> component is also being used in Transactions Page , BatchTxList component

Transactions page takes 80% width of screen , it would have enough space in most cases
As suggested 
Applied truncateText as default to handle very large function names